### PR TITLE
Add plugin: yearly-diary-comparator

### PR DESCRIPTION
**Pull request by kiitosu**
# I am submitting a new Community Plugin
This plugin provides a yearly calendar view for Obsidian daily notes, allowing you to compare your activities or diary entries across different years. By displaying entries for the same date side by side, you can intuitively review changes and track your personal growth year over year.

## Repo URL

Link to my plugin: https://github.com/kiitosu/yearly-diary-comparator

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by kiitosu** [comment]
@manaoumov  
Thank you very much for your review!

I've revised the code and updated the release assets.  
Could you please take a look?

- https://github.com/kiitosu/yearly-diary-comparator  
- https://github.com/kiitosu/yearly-diary-comparator/releases

**Comment by joethei** [comment]
[.table-wrapper {](https://github.com/kiitosu/yearly-diary-comparator/blob/1c59ffbdc61426f3cbd3dc6cf5d29f249c3400f5/styles.css#L66-L111)
These class name are too generic.

[// styles.cssをheadに追加](https://github.com/kiitosu/yearly-diary-comparator/blob/1c59ffbdc61426f3cbd3dc6cf5d29f249c3400f5/main.ts#L78-L93), [// headからstyle要素を削除](https://github.com/kiitosu/yearly-diary-comparator/blob/1c59ffbdc61426f3cbd3dc6cf5d29f249c3400f5/main.ts#L141-L146)
This is not necessary, Obsidian already handles that for you.
Remove this code.

[leaves[0].detach();](https://github.com/kiitosu/yearly-diary-comparator/blob/1c59ffbdc61426f3cbd3dc6cf5d29f249c3400f5/main.ts#L103)
Do not detach leaves with your custom view here, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>
Instead, only create a new leaf, if there is not one already.
If there is a leaf, focus on it.

**Comment by kiitosu** [comment]
@joethei 
Thank you for your review!
I’ve addressed the issues you pointed out. Please let me know if anything else needs to be updated.
- https://github.com/kiitosu/yearly-diary-comparator/releases
